src/CodeGen_Xtensa_vectors.template.cpp

```diff
@@ -3112,6 +3112,18 @@ convert<native_vector_u32_x4, native_vector_i24>(const native_vector_i24 &src) {
                                 IVP_CVT32S2NX24HL(src), IVP_CVT32S2NX24HH(src));
 }
 
+HALIDE_ALWAYS_INLINE native_vector_i32_x2
+halide_xtensa_div_i32_i16(native_vector_i32_x2 a, native_vector_i16 b) {
+    native_vector_i32 reminder, quotent1, quotent2;
+    native_vector_i32_x2 b_i32 = convert<native_vector_i32_x2, native_vector_i16>(b);
+
+    IVP_DIVN_2X32X16(quotent1, reminder, a.native_vector[0], IVP_MOVNX16_FROMN_2X32(b_i32.native_vector[0]), 0);
```

Comment:
Initially I was very confused about what's happening here, but it does make sense now: it looks like IVP_DIVN_2X32X16 expects 16-bit divisor to be interleaved and the last argument (which can be 0 or 1) controls whether even or odd lanes are taken.  In this function, by converting to 32-bit and then reinterpreting it as 16 bit , we sort of make all of the elements to be in even positions (thus the last argument is 0 in both calls). 

I think the more efficient (and in my opinion more readable) way might be to stay in 16-bit, but pre-interleave the vector b with something like IVP_SELNX16I(b, b, IVP_SELI_16B_INTERLEAVE_1_EVENODD)?

Reply:
Done